core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -0,0 +1,138 @@
+        Row(database, tableName, isTemp, s"$information\n")
+      } else {
+        Row(database, tableName, isTemp)
+      }
+    }
+  }
+}
+
+case class TiDescribeTablesCommand(tiContext: TiContext, delegate: DescribeTableCommand)
+    extends TiCommand(delegate) {
+  override val output: Seq[Attribute] = Seq(
+    // Column names are based on Hive.
+    AttributeReference(
+      "col_name",
+      StringType,
+      nullable = false,
+      new MetadataBuilder().putString("comment", "name of the column").build()
+    )(),
+    AttributeReference(
+      "data_type",
+      StringType,
+      nullable = false,
+      new MetadataBuilder().putString("comment", "data type of the column").build()
+    )(),
+    AttributeReference(
+      "nullable",
+      StringType,
+      nullable = false,
+      new MetadataBuilder().putString("comment", "whether the column is nullable").build()
+    )(),
+    AttributeReference(
+      "comment",
+      StringType,
+      nullable = true,
+      new MetadataBuilder().putString("comment", "comment of the column").build()
+    )()
+  )
+
+  override def run(sparkSession: SparkSession): Seq[Row] =
+    tiCatalog
+      .catalogOf(delegate.table.database)
+      .getOrElse(
+        throw new NoSuchDatabaseException(
+          delegate.table.database.getOrElse(tiCatalog.getCurrentDatabase)
+        )
+      ) match {
+      case _: TiSessionCatalog =>
+        val result = new ArrayBuffer[Row]
+        if (delegate.partitionSpec.nonEmpty) {
+          throw new AnalysisException(
+            s"DESC PARTITION is not supported on TiDB table: ${delegate.table.identifier}"
+          )
+        }
+        val metadata = tiCatalog.getTableMetadata(delegate.table)
+        describeSchema(metadata.schema, result, header = false)
+
+        if (delegate.isExtended) {
+          describeFormattedTableInfo(metadata, result)
+        }
+
+        result
+      case _: SessionCatalog => super.run(sparkSession)
+    }
+
+  private def describeSchema(schema: StructType,
+                             buffer: ArrayBuffer[Row],
+                             header: Boolean): Unit = {
+    if (header) {
+      append(buffer, s"# ${output.head.name}", output(1).name, output(2).name)
+    }
+    schema.foreach { column =>
+      append(
+        buffer,
+        column.name,
+        column.dataType.simpleString,
+        column.getComment().orNull,
+        column.nullable.toString
+      )
+    }
+  }
+
+  private def append(buffer: ArrayBuffer[Row],
+                     column: String,
+                     dataType: String,
+                     comment: String,
+                     nullable: String = ""): Unit =
+    buffer += Row(column, dataType, nullable, comment)
+
+  private def describeFormattedTableInfo(table: CatalogTable, buffer: ArrayBuffer[Row]): Unit = {
+    // So far we only have engine name and primary key for extended information.
```

Comment:
What exactly is engine name here? Like BTree?

Reply:
Oh, that was not yet done in this PR. I will delete them.